worker/src/lib.rs

```diff
@@ -299,7 +300,73 @@ pub async fn witness_discord(
             vc.generate_proof(
                 &sk,
                 &LinkedDataProofOptions {
-                    verification_method: Some(format!("{}#controller", SPRUCE_DIDWEB)),
+                    verification_method: Some(URI::String(format!("{}#controller", SPRUCE_DIDWEB))),
+                    ..Default::default()
+                }
+            )
+            .await
+        );
+        vc.proof = Some(OneOrMany::One(proof));
+
+        Ok(jserr!(serde_json::to_string(&vc)).into())
+    })
+}
+
+#[wasm_bindgen]
+pub async fn dns_lookup(
+    secret_key_jwk: String,
+    public_key_tezos: String,
+    domain: String,
+) -> Promise {
+    initialize_logging();
+
+    future_to_promise(async move {
+        let pk: JWK = jserr!(jwk_from_tezos_key(&public_key_tezos));
+        let sk: JWK = jserr!(serde_json::from_str(&secret_key_jwk));
+
+        let dns_result = jserr!(dns::retrieve_dns_response(domain.clone()).await);
+
+        let mut vc = jserr!(dns::build_dns_vc(&pk));
+
+        let mut signature_to_resolve = "".to_string();
```

Comment:
That's minor but in Rust we generally avoid mutable variables at all cost. So sorting through a list would be put in a function